lib/src/di/injector.dart

```diff
@@ -99,19 +113,30 @@ void _injectApiDependencies() {
 }
 
 void _injectNotifiers() {
-  _locator.registerFactory(() => ThemeNotifier(_locator<SettingsRepository>()));
+  _locator.registerSingleton<HomePageNotifier>(HomePageNotifier());
+  _locator.registerSingleton<ThemeNotifier>(ThemeNotifier(_locator<SettingsRepository>()));
```

Comment:
to może zostać fabryką, bo nie przechowuje stanu

Reply:
a właśnie, tak się zacząłem zastanawiać, czy przypadkiem wszystkie notifiery nie powinny być fabrykami i być tworzone tylko przy tworzeniu widgeta (strony) dla której są dedykowane? I że np. poprzez to że parę razy je injectujemy dla wygody to robimy źle, a singleton tylko maskuje ten błąd (bo gdyby były fabrykami to byśmy niepotrzebnie tworzyli kolejne instancje)? 🤔 wiesz, viewmodel czy presenter też trzymają stan i nie są singletonami ;)

Reply:
Jedynym przykładem jaki widzę dla notifiera - singletona jest case, gdzie dzieli stan między więcej niż jednym widokiem. Wtedy trzeba wybrać czy providować takiego notifiera poziom wyżej czy zrobić z niego singletona. Zawsze robiłem to jako singleton, ale rzeczywiście nie badałem która z opcji jest lepsza, albo czy jest coś co jedną z nich wyklucza. Też notifier to nie view model, trochę inną ma zasadę działania. U nas w apce nie ma takiego casa, więc wszystkie notifiery powinny być tworzone przez fabrykę.

Reply:
No właśnie tak zacząłem się zastanawiać, bo jeżeli jest to singleton to np. już raz byliśmy na danym ekranie, wyszliśmy z niego, nie ma go na stacku, potem przechodzimy znów na niego i stan notifiera jest już określony na podstawie tego co działo się wcześniej - przykładowo wczesniej coś tam napisałeś w textfieldzie i notifier niepotrzebnie to przechowuje. Z kolei jak notifier jest wykorzystywany w więcej niż jednym miejscu to jak trzyma w sobie jakieś dane z ui, których nie wysłał na backend, to żeby mieć sharowany changenotifier trzeba pilnować, żeby tylko raz wstrzyknąć, bo wtedy będziemy mieli różne instancje z różnymi stanami, ale tak czy siak wydaje mi się, że to lepsze niż singleton :thinking:

Reply:
>to żeby mieć sharowany changenotifier trzeba pilnować, żeby tylko raz wstrzyknąć
to już get_it ogarnia, żeby zwracać zawsze tą samą instancję przy wstrzykiwaniu

Reply:
może nieprecyzyjnie napisałem, chodziło mi o przypadek kiedy sharowany notifier jest factorką :)